src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/Terminal.component.tsx

```diff
@@ -1,41 +1,119 @@
-import React, { useEffect, useState } from 'react'
-import Tippy from '@tippyjs/react'
-import { ReactComponent as Disconnect } from '../../../../assets/icons/ic-disconnect.svg'
-import { ReactComponent as Connect } from '../../../../assets/icons/ic-connect.svg'
-import { ReactComponent as Abort } from '../../../../assets/icons/ic-abort.svg'
+import React, { useEffect, useRef, useState } from 'react'
 import { useParams, useRouteMatch } from 'react-router'
 import { NodeDetailTab } from '../nodeDetail.type'
 import IndexStore from '../../../index.store'
-import Select from 'react-select'
 import { SocketConnectionType } from './node.type'
-import TerminalView from './terminal/Terminal'
 import MessageUI from '../../../../common/message.ui'
 import { Option } from '../../../../common/ReactSelect.utils'
-import { getContainersData, getContainerSelectStyles, getGroupedContainerOptions, getShellSelectStyles } from '../nodeDetail.util'
+import {
+    getContainersData,
+    getContainerSelectStyles,
+    getGroupedContainerOptions,
+    getShellSelectStyles,
+} from '../nodeDetail.util'
 import { shellTypes } from '../../../../../../config/constants'
 import { OptionType } from '../../../../../app/types'
-import { Options, TerminalComponentProps } from '../../../appDetails.type'
+import { AppType, Options, TerminalComponentProps } from '../../../appDetails.type'
 import './nodeDetailTab.scss'
+import TerminalWrapper from './terminal/TerminalWrapper.component'
+import { TerminalSelectionListDataType } from './terminal/terminal.type'
+import { get, showError } from '@devtron-labs/devtron-fe-common-lib'
+
+let clusterTimeOut
 
 function TerminalComponent({
     selectedTab,
     isDeleted,
     isResourceBrowserView,
     selectedResource,
+    selectedContainer,
+    setSelectedContainer,
 }: TerminalComponentProps) {
     const params = useParams<{ actionName: string; podName: string; nodeType: string; node: string }>()
     const { url } = useRouteMatch()
+    const terminalRef = useRef(null)
     const podMetaData = !isResourceBrowserView && IndexStore.getMetaDataForPod(params.podName)
```

Comment:
these should be derived from props to reduce checks

Reply:
done